### PR TITLE
fix string can be used for stringlist

### PR DESCRIPTION
**Pull request by Gnouc**

### CODE REVIEW

**Comment by Gnouc** [comment]
cc @tonioo

File: sievelib/commands.py [hunk 1]
```diff
@@ -347,6 +347,19 @@ def __is_valid_value_for_arg(self, arg, value):
             return True
         return value.lower() in arg["values"]
 
+    def __is_valid_type(self, typ, typlist):
+        """ Check if type is valid base on input type list
```
Comment (tonioo) [inline sievelib/commands.py]: based on

File: sievelib/commands.py [hunk 2]
```diff
@@ -347,6 +347,19 @@ def __is_valid_value_for_arg(self, arg, value):
...
+            "string" is special because it can be used for stringlist
+
+        :param typ: the type to check
+        :param typlist: the list of type to check
+        :return: True on success, False otherwise
+        """
+        typ_is_str = typ == "string"
+        str_list_in_typlist = "stringlist" in typlist
+
+        return typ not in typlist and not (typ_is_str and str_list_in_typlist)
```
Comment (tonioo) [inline sievelib/commands.py]:
```suggestion
        return typ in typlist or (typ_is_str and str_list_in_typlist)
```
Looks easier to understand.

File: sievelib/commands.py [hunk 3]
```diff
@@ -405,7 +418,7 @@ def check_next_arg(self, atype, avalue, add=True, check_extension=True):
                         if not curarg["name"] in self.arguments:
                             self.arguments[curarg["name"]] = []
                         self.arguments[curarg["name"]] += [avalue]
-                elif atype not in curarg["type"] or \
+                elif self.__is_valid_type(atype, curarg["type"]) or \
```
Comment (tonioo) [inline sievelib/commands.py]:
```suggestion
                elif not self.__is_valid_type(atype, curarg["type"]) or \
```
According to my previous comment.

**Comment by Gnouc** [comment]
@tonioo Updated

**Comment by tonioo** [comment]
@Gnouc Thank you!

**Pull request merged**